analysis/deep2_r23_o32/plotting/plot_difference_curvefit.py

```diff
@@ -0,0 +1,284 @@
+    an_txt += r'$\tilde\Delta_{R_{23}}$ : %0.2f' % med0 + '\n'
+    an_txt += r'$\sigma$ : %0.2f' % sig0
+    ax.set_ylabel(r'$\Delta_{R_{23}} \equiv \log(R_{23}) '
+                  r'- \log(R_{23})_{\rm Zcal_curvefit}$')
+
+    ax.annotate(an_txt, [0.2, 0.015], xycoords='axes fraction',
+                va='bottom', ha='right', fontsize=10)
+    ax.set_xlabel(r'$12+\log({\rm O/H})_{T_e}$')
+
+    ax.minorticks_on()
+
+    if len(OH_range) != 0:
+        ax.set_xlim(OH_range)
+    if len(data_range) != 0:
+        ax.set_ylim(data_range)
+    ax.set_xlim(7, 9.5)
+    leg_R23 = ax.legend(loc='upper right', scatterpoints=1, fontsize=8,
+                            framealpha=0.5)
+    for lh in leg_R23.legendHandles:
+        lh.set_alpha(0.5)
+
+    plt.subplots_adjust(left=0.12, right=0.97, bottom=0.1, top=0.97)
+
+    # log.info(f"Writing: {pdf_file}")
+    fig.savefig(pdf_file)
+
+    # log.info("finished.")
+
+
+def plot_difference_twovariable(lR23, lO32, OH, bin_start, bin_end, pdf_file,
+                                new_coefficients=[], n_bins=4, data_err=[],
+                                OH_err=[], OH_range=[], data_range=[-0.3, 0.3],
+                                marker=[], label=[], IDs=[], log=None):
+    """
+    Plot differences between LACR23 vs observed R23 as a function
+    of metallicity
+    Only going to be working for a comparison of lR23 to metallicity with
+    color binning based on lO32 like the three variable plots
+    -> maybe eventually I will add the lO32 comparison back, but we haven't
+    been using it
+    Used by main() function
+    """
+
+    if log is None:
+        log = log_stdout()
+
+    log.info("starting ...")
+
+    fig, ax = plt.subplots()
+
+    n_sample = len(lR23)
+
+    # Plotting
+    if len(marker) == 0:
+        marker = ['o'] * n_sample
+
+    diff0 = []
+    for nn in range(n_sample):
+        if len(new_coefficients) != 0:
+            LAC = bian18_R23_OH(OH[nn], new_coefficients)
+            log.info(f"curve fit LAC_R23: {LAC}")
+        else:
+            LAC = bian18_R23_OH(OH[nn], bian_coeff)
+            log.info(f"LAC_R23: {LAC}")
+
+        if nn == 0:
+            if IDs:
+                for jj in range(len(LAC)):
+                    id_diff = lR23[0][jj]-LAC[jj]
+                    ax.annotate(IDs[0][jj], (OH[0][jj], id_diff),
+                                fontsize='6')
+
+        # Label in upper left the points
+        if len(label) != 0:
+            x1 = OH_range[0] + 0.025 * (OH_range[1] - OH_range[0])
+            y1 = data_range[1] - (nn * 0.035 + 0.05) \
+                 * (data_range[1] - data_range[0])
+            x2 = OH_range[0] + 0.035 * (OH_range[1] - OH_range[0])
+            y2 = data_range[1] - (nn * 0.035 + 0.0525) \
+                 * (data_range[1] - data_range[0])
+            ax.text(x2, y2, label[nn], fontsize=8, va='center', ha='left')
+            ax.plot([x1], [y1], marker=marker[nn], color='black')
```

Comment:
You seem to have this repeated so perhaps best to refactor into another function for multiple modules

Reply:
done